datashader/datatypes.py

```diff
@@ -0,0 +1,361 @@
+import numpy as np
+from pandas.api.extensions import ExtensionDtype, ExtensionArray
+from pandas.core.dtypes.dtypes import register_extension_dtype
```

Comment:
This should be imported from `pandas.api.extensions` as well. Pandas considers `pandas.core` to be private.